pkg/kubelet/cm/memorymanager/memory_manager_test.go

```diff
@@ -18,24 +18,112 @@ package memorymanager
 
 import (
 	"fmt"
+	"io/ioutil"
+	"os"
 	"reflect"
 	"strings"
 	"testing"
 
-	info "github.com/google/cadvisor/info/v1"
+	cadvisorapi "github.com/google/cadvisor/info/v1"
 	"github.com/stretchr/testify/assert"
-
 	v1 "k8s.io/api/core/v1"
 	"k8s.io/apimachinery/pkg/api/resource"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/types"
+	runtimeapi "k8s.io/cri-api/pkg/apis/runtime/v1alpha2"
+	"k8s.io/kubernetes/pkg/kubelet/cm/containermap"
+	"k8s.io/kubernetes/pkg/kubelet/cm/memorymanager/state"
+	"k8s.io/kubernetes/pkg/kubelet/cm/topologymanager"
 )
 
 const (
 	hugepages2M = "hugepages-2Mi"
 	hugepages1G = "hugepages-1Gi"
 )
 
+type testMemoryManager struct {
+	description                string
+	policy                     Policy
+	machineInfo                cadvisorapi.MachineInfo
+	assignments                state.ContainerMemoryAssignments
+	expectedAssignments        state.ContainerMemoryAssignments
+	machineState               state.NodeMap
+	expectedMachineState       state.NodeMap
+	expectedError              error
+	expectedAllocateError      error
+	expectedAddContainerError  error
+	updateError                error
+	removeContainerID          string
+	nodeAllocatableReservation v1.ResourceList
+	policyName                 string
+	affinity                   topologymanager.Store
+	preReservedMemory          map[int]map[v1.ResourceName]resource.Quantity
+	expectedHints              map[string][]topologymanager.TopologyHint
+	expectedReserved           systemReservedMemory
+	reserved                   systemReservedMemory
+	podAllocate                *v1.Pod
+	firstPod                   *v1.Pod
+	activePods                 []*v1.Pod
+}
+
+func returnPolicyByName(testCase testMemoryManager) Policy {
+	switch testCase.policyName {
+	case "mock":
+		return &mockPolicy{
+			err: fmt.Errorf("Fake reg error"),
+		}
+	case "static":
```

Comment:
can use please use constants for policies names, we already have 
`policyTypeStatic`
`policyTypeNone`
and you can add `policyTypeMock`, just to avoid magic strings

Reply:
Done